components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java

```diff
@@ -4319,21 +4299,64 @@ public static ApiEndpointValidationResponseDTO sendHttpHEADRequest(String urlVal
         return apiEndpointValidationResponseDTO;
     }
 
-    private boolean validateServiceCatalogEntries(LinkedHashMap endpointConfig) {
-        Object productionCatalogEntry = null;
-        Object sandboxCatalogEntry = null;
-        LinkedHashMap productionEps = (LinkedHashMap) endpointConfig.get("production_endpoints");
-        LinkedHashMap sandboxEps = (LinkedHashMap) endpointConfig.get("sandbox_endpoints");
-        if (productionEps == null && sandboxEps == null) {
+    private void validateServiceCatalogEntries(LinkedHashMap endpointConfig) {
+
+        Map<String, Boolean> productionEndpointStatus =
+                validateServiceCatalogEndpoint(endpointConfig, APIConstants.API_DATA_PRODUCTION_ENDPOINTS);
+        Map<String, Boolean> sandboxEndpointStatus =
+                validateServiceCatalogEndpoint(endpointConfig, APIConstants.API_DATA_SANDBOX_ENDPOINTS);
+        Map<String, Boolean> productionFailoverStatus =
+                validateServiceCatalogEndpoint(endpointConfig, APIConstants.ENDPOINT_PRODUCTION_FAILOVERS);
+        Map<String, Boolean> sandboxFailoverStatus =
+                validateServiceCatalogEndpoint(endpointConfig, APIConstants.ENDPOINT_SANDBOX_FAILOVERS);
+
+        if (!productionEndpointStatus.get("hasEndpoints") && !sandboxEndpointStatus.get("hasEndpoints")) {
             RestApiUtil.handleBadRequest("At least one endpoint from Production Endpoint or " +
                     "Sandbox Endpoint must be defined.", log);
         }
-        if (productionEps != null) {
-            productionCatalogEntry = productionEps.get("id");
+        if (!(productionEndpointStatus.get("validCatalogEndpoints") &&
```

Comment:
since "validCatalogEndpoints" is used repetitively, shall we add it to a constant field?

Reply:
inside the method?

Reply:
to a constant class

Reply:
Fixed in https://github.com/wso2/carbon-apimgt/pull/8728/commits/216e08f31573e70a482b6f9e3235b139d06c07bf